test/integration/default/default_spec.rb

```diff
@@ -400,3 +400,13 @@
     its('gid') { should eq 7000 }
   end
 end
+
+describe user('system_user') do
+  it { should exist }
+  its('uid') { should be < 1000 } unless os_family == 'darwin'
```

Comment:
Does MacOS put system users in a different number range that we should be checking for?

Reply:
@ramereth  https://github.com/chef/chef/blob/master/lib/chef/provider/user/mac.rb starts system uids at 200 but in my test it is above 500 despite system being set. I think it is too unreliable to test. The https://github.com/chef/chef/blob/master/lib/chef/provider/group/dscl.rb doesnt do anything with the system option either.